### PR TITLE
Refactor some small uses of DiscoveryNodes version methods

### PR DESCRIPTION
**Pull request by thecoop**
Refactor a couple of small uses of version-based methods

### CODE REVIEW

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/AbstractProfilingPersistenceManager.java [hunk 1]
```diff
@@ -71,6 +73,18 @@ public void setTemplatesEnabled(boolean templatesEnabled) {
         this.templatesEnabled = templatesEnabled;
     }
 
+    private static boolean isMixedVersionCluster(DiscoveryNodes nodes) {
+        Version version = null;
+        for (var n : nodes) {
+            if (version == null) {
+                version = n.getVersion();
+            } else if (version.equals(n.getVersion()) == false) {
+                return false;
```
Comment (danielmitterdorfer) [inline x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/AbstractProfilingPersistenceManager.java:82]: Shouldn't this return `true`? i.e. if the there is a version mismatch in any two nodes we infer a mixed version cluster?

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/AbstractProfilingPersistenceManager.java [hunk 2]
```diff
@@ -71,6 +73,18 @@ public void setTemplatesEnabled(boolean templatesEnabled) {
...
+            }
+        }
+        return true;
```
Comment (danielmitterdorfer) [inline x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/AbstractProfilingPersistenceManager.java:85]: Shouldn't this return `false`?

File: x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/AbstractProfilingPersistenceManager.java (hunk 1 above)
Comment (thecoop) [reply x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/AbstractProfilingPersistenceManager.java:82]: Doh, wrong way round...

**Review by danielmitterdorfer — APPROVED** [review]
LGTM, thanks!

**Pull request merged**